### PR TITLE
Update targets to new listing method

### PR DESCRIPTION
**Pull request by jefferai**
This uses the prefetch method to fetch resource IDs and validate them
for authz by the user before then fetching the full list of targets.

### CODE REVIEW

File: internal/target/repository.go [hunk 1]
```diff
@@ -138,21 +141,98 @@ func (r *Repository) LookupTarget(ctx context.Context, publicIdOrName string, op
...
+	var numRows int
+	for rows.Next() {
+		numRows++
+		var tv targetView
+		if err := r.reader.ScanRows(ctx, rows, &tv); err != nil {
+			return nil, errors.Wrap(ctx, err, op, errors.WithMsg("scan row failed"))
+		}
+		targetsMap[tv.GetScopeId()] = append(targetsMap[tv.GetScopeId()], tv)
+	}
+
```
[the remark this answers is not in the document]
Comment (johanbrandhorst) [reply internal/target/repository.go:189]: Apparently `ScanRows` checks the iteration error already

File: internal/session/repository_session.go [hunk 2]
```diff
@@ -263,38 +264,38 @@ func (r *Repository) ListSessions(ctx context.Context, opt ...Option) ([]*Sessio
 	case 0:
 	case 1:
 		inClauseCnt += 1
-		where, args = append(where, fmt.Sprintf("scope_id = @%d", inClauseCnt)), append(args, sql.Named(fmt.Sprintf("%d", inClauseCnt), opts.withScopeIds[0]))
+		where, args = append(where, fmt.Sprintf("scope_id = @%d", inClauseCnt)), append(args, sql.Named(strconv.Itoa(inClauseCnt), opts.withScopeIds[0]))
```
Comment (johanbrandhorst) [inline internal/session/repository_session.go:267]: Hm, I made these changes based on my comment, but I see this pattern is pretty common in this file, so maybe I should revert it again. Thoughts @tmessi?

Comment (tmessi) [reply internal/session/repository_session.go:267]: Yea I have noticed this in a lot of the repositories. I think this should be changed, but probably as a separate commit that addresses it in all places, so for now I would revert this.

Comment (johanbrandhorst) [reply internal/session/repository_session.go]: https://github.com/hashicorp/boundary/pull/2047

**Review by tmessi — APPROVED** [review]

**Pull request merged**